src/main/java/teammates/storage/api/AdminEmailsDb.java

```diff
@@ -225,7 +229,11 @@ public AdminEmailAttributes getAdminEmail(String subject, Date createDate){
         
         Iterator<AdminEmail> it = adminEmailList.iterator();
         while(it.hasNext()){
-            list.add(new AdminEmailAttributes(it.next()));
+            AdminEmail adminEmail = it.next();
+            
+            if (!JDOHelper.isDeleted(adminEmail)) {
```

Comment:
Let's remove that check for <code>isDeleted</code> on line 226 then. That check looks wrong.